src/composition-engine.js

```diff
@@ -69,6 +69,37 @@ function tryActivateViewModel(context) {
   return context.viewModel.activate(context.model) || Promise.resolve();
 }
 
+class SwapStrategies {
+  // animate the next view in before removing the current view;
+  before(viewSlot, previousView, callback) {
+    let promise = Promise.resolve(callback());
+
+    if (previousView !== undefined) {
+      return promise.then(() => viewSlot.remove(previousView, true));
+    }
+
+    return promise;
+  }
+
+  // animate the next view at the same time the current view is removed
+  with(viewSlot, previousView, callback) {
+    let promise = Promise.resolve(callback());
+
+    if (previousView !== undefined) {
+      return Promise.all([viewSlot.remove(previousView, true), promise]);
+    }
+
+    return promise;
+  }
+
+  // animate the next view in after the current view has been removed
+  after(viewSlot, previousView, callback) {
+    return Promise.resolve(viewSlot.removeAll(true)).then(callback);
+  }
+}
+
+const swapStrategies = new SwapStrategies();
```

Comment:
Do we actually need a class? Seems overkill as this is just a collection of named methods. Is there ever going to be a second instance?

Also, per my previous comment about duplication in `templating-router`, I suppose we would need to `export const swapStrategies`?